internal/envoy/runtime.go

```diff
@@ -57,6 +58,31 @@ type Runtime struct {
 	shutdownHooks []func(context.Context) error
 }
 
+// String is only used in tests. It is slow, but helps when debugging CI failures
```

Comment:
@mathetake let me know what you think about this.. this is only for tests (like TestFuncERun) which are in a different package or invoked via urfave which makes it impossible to print out the state here on fail. It is some extra code, but it is tested.

Reply:
In order to achieve this we will have to define interface over Runtime vs the codebase complexity. I'm fine for now with this, we can come back to this in another PR if we want to move out this test only function😄

Reply:
yeah let's do a refactoring PR later if you are ok with it for an hour :D because there's some package complexity due to some things being in cmd_test vs cmd etc

Reply:
https://github.com/tetratelabs/func-e/pull/349